pkg/services/ngalert/metrics/historian.go

```diff
@@ -7,17 +7,59 @@ import (
 )
 
 type Historian struct {
-	WriteDuration *instrument.HistogramCollector
+	Info              *prometheus.GaugeVec
+	TransitionsTotal  *prometheus.CounterVec
+	TransitionsFailed *prometheus.CounterVec
+	WritesTotal       *prometheus.CounterVec
+	WritesFailed      *prometheus.CounterVec
+	WriteDuration     *instrument.HistogramCollector
+	BytesWritten      prometheus.Counter
 }
 
 func NewHistorianMetrics(r prometheus.Registerer) *Historian {
 	return &Historian{
+		Info: promauto.With(r).NewGaugeVec(prometheus.GaugeOpts{
+			Namespace: Namespace,
+			Subsystem: Subsystem,
+			Name:      "state_history_info",
+			Help:      "Information about the state history backend.",
+		}, []string{"backend"}),
+		TransitionsTotal: promauto.With(r).NewCounterVec(prometheus.CounterOpts{
+			Namespace: Namespace,
+			Subsystem: Subsystem,
+			Name:      "state_history_transitions_total",
+			Help:      "The total number of state transitions processed by the state historian.",
+		}, []string{"org"}),
+		TransitionsFailed: promauto.With(r).NewCounterVec(prometheus.CounterOpts{
+			Namespace: Namespace,
+			Subsystem: Subsystem,
+			Name:      "state_history_transitions_failed_total",
+			Help:      "The total number of state transitions that failed to be written - they are not retried.",
+		}, []string{"org"}),
+		WritesTotal: promauto.With(r).NewCounterVec(prometheus.CounterOpts{
+			Namespace: Namespace,
+			Subsystem: Subsystem,
+			Name:      "state_history_writes_total",
+			Help:      "The total number of state history batches that were attempted to be written.",
+		}, []string{"org"}),
+		WritesFailed: promauto.With(r).NewCounterVec(prometheus.CounterOpts{
+			Namespace: Namespace,
+			Subsystem: Subsystem,
+			Name:      "state_history_writes_failed_total",
+			Help:      "The total number of failed writes of state history batches.",
+		}, []string{"org"}),
 		WriteDuration: instrument.NewHistogramCollector(promauto.With(r).NewHistogramVec(prometheus.HistogramOpts{
 			Namespace: Namespace,
 			Subsystem: Subsystem,
 			Name:      "state_history_request_duration_seconds",
 			Help:      "Histogram of request durations to the state history store.",
 			Buckets:   instrument.DefBuckets,
 		}, instrument.HistogramCollectorBuckets)),
+		BytesWritten: promauto.With(r).NewCounter(prometheus.CounterOpts{
+			Namespace: Namespace,
+			Subsystem: Subsystem,
+			Name:      "state_history_writes_bytes_total",
+			Help:      "The total number of bytes sent within a batch to the state history store.",
+		}),
```

Comment:
Did we missed per org here?

Reply:
A histogram is an abstraction on top of counters - for each histogram, you have 2 + N counters, where N is the number of defined buckets. The reason why we discourage the use of histogram per tenant is due to the cardinality explosion of series for `histogram = (2 + N) x tenants`.

For bytes, it's pretty harmless and I think there's usefulness in having granularity in total egress costs per tenant (if I were operating Grafana).

For real though, it's such a meaningless nit - I'm happy if you want to do that in a separate PR or if you want to not go down this route.